uni/UAlberta.py

```diff
@@ -239,26 +239,28 @@ def scrapeCourseList(self, conn, termid):
             else:
                 courseList['teachers'] = ["N/A"]
 
-            # for entry in list, match the days, startTime, endTime, and locations to course
-            for entry_times in times_list:
+            # Get a list of times and locations associated with a course
+            times = [x for x in times_list if x['attributes']['class'] == courseList['id']]
 
-                if entry_times['attributes']['class'] == courseList['id']:
+            for entry_time in times:
 
-                    # Combines day, startTime, endTime into a duration
-                    duration = " "
-                    duration = duration.join(
-                        (entry_times['attributes']['day'][0], entry_times['attributes']['startTime'][0].replace(" ", ""),
-                         entry_times['attributes']['endTime'][0].replace(" ", "")))
+                times_list.remove(entry_time)
 
-                    # Adds '-' btw the times
-                    duration = re.sub(r'^((.*?\s.*?){1})\s', r'\1 - ', duration)
-                    courseList['times'] = [duration]
+                # Combines day, startTime, endTime into a duration
+                duration = " "
+                duration = duration.join(
+                    (entry_time['attributes']['day'][0], entry_time['attributes']['startTime'][0].replace(" ", ""),
```

Comment:
To prevent the redundant `attribute` key, we should store a variable such as `attributes = entry_times['attributes']` above and use it here to clean up.